app/models/concerns/hyku_addons/user_behavior.rb

```diff
@@ -4,30 +4,40 @@ module HykuAddons
   module UserBehavior
     extend ActiveSupport::Concern
 
+    PROFILE_VISIBILITY = { open: "open", closed: "closed" }.freeze
+
     included do
-      around_update :toggle_display_profile
-      validate :must_have_valid_email_format
-    end
+      before_save :toggle_display_profile
 
-    def must_have_valid_email_format
-      accepted_email_formats = current_account&.settings&.dig('email_format')
-      return unless accepted_email_formats.present?
+      validate :email_format
 
-      email_format = '@' + email.split('@')[-1]
-      errors.add(:email, "Email must contain #{accepted_email_formats.join(', ')}") unless accepted_email_formats.include? email_format
+      scope :with_public_profile, -> { where(display_profile: true) }
     end
 
+    # NOTE: Is this used anywhere?
     def current_account
       Site.account
     end
 
     def display_profile_visibility
-      display_profile ? "open" : "closed"
+      PROFILE_VISIBILITY[display_profile ? :open : :closed]
+    end
+
+    protected
+
+    # Triggered when the user registers an account
+    def email_format
+      email_formats = Site.account&.settings&.dig("email_format")
+
+      return if email_formats.blank? || email_formats.include?("@#{email.split("@").last}")
```

Comment:
This seems a little obscure.  Can't we just use a regex to check the email is in the right format? This just checks it is an @ symbol followed by at least one character, which might not be alphanumeric.  We don't even know whether it has a domain. Are the email_formats just a list of domains the tenant's user's emails must belong to?

Reply:
@thenapking I would like to have done this with the Rails validation `format`, so i'm seeing if thats possible, but i'm not sure it is given the error message and moving all this code into a lamba on the `validate` call doesn't seem better to me.

Reply:
How do we check the validity of the email_formats? Could you link me to that piece of code? My feeling is that it would be better to store email_domains, not formats, then check that the email includes one of those domains when split at the @ symbol.  But email validation is always tricky and we seem to be rolling our own here for no reason.

Reply:
@thenapking normally i would agree that this could be tweaked, however at this stage, maintaining the functionality is my primary concern because we need to work out a migration for existing clients could be required. 

But yes, storing the `@` and the domain is excessive.